runtime/etc/src/mli/client_runtime.c

```diff
@@ -119,7 +126,8 @@ void chpl_library_init(int argc, char** argv) {
     chpl_mli_debugf("Passing along arg %d: %s\n", i, argv[i]);
     argv_plus_sock[i] = argv[i];
   }
-  argv_plus_sock[argc] = "--chpl-mli-socket-loc";
+  char socketFlag[22] = "--chpl-mli-socket-loc";
```

Comment:
Is the length necessary here? Just curious as to why we use an array with a magic number over a `const char*`.

Reply:
The error I was getting was due to putting a `const char[22]` into a `char*`.  Since argv needed to be populated based on what was already present, I figured a `const char**` wasn't appropriate.  Alternatively, I could cast the literal into a `char *`.  I wasn't sure which was the better strategy

Reply:
Totally slipped my mind that argv is not const, sounds good!